### PR TITLE
Handle GQL error response

### LINKED ISSUE #464: Better error handling if startRepositoryMigration fails
Recently startRepositoryMigration endpoint implemented some pre-migration validation. If the validation fails (e.g. workflow scope missing), the API call to startRepositoryMigration will not return a migration ID and will instead return an error message (but still an HTTP 200).

Right now this will cause the CLI to crash, because when it gets a 200 response it expects to find a migration ID.

It would be better to look for the error message in the response, and raise an OctoshiftCliException.

You can see an example of this failure here: https://github.com/github/gh-gei/runs/6995914753?check_suite_focus=true#step:17:168

### PR DESCRIPTION
**Pull request by ArinGhazarian**
Fixes #464 

- [x] Did you write/update appropriate tests
- [x] Release notes updated (if appropriate)
- [x] Appropriate logging output
- [x] Issue linked
- [x] Docs updated (or issue created)

### Screenshot 
image

### CODE REVIEW

**Comment by dylan-smith** [comment]
should this PR be non-draft now?

**Comment by dylan-smith** [comment]
Could you attach a screenshot of calling migrate-repo with  a PAT without workflow scope

File: src/Octoshift/GithubApi.cs [hunk 1]
```diff
@@ -617,5 +619,16 @@ private static Mannequin BuildMannequin(JToken mannequin)
...
         }
+
+        private void EnsureSuccessGraphQLResponse(JObject response)
+        {
+            if (response.TryGetValue("errors", out var jErrors) && jErrors is JArray { Count: > 0 } errors)
+            {
+                var error = (JObject)errors[0];
+                var errorType = error.TryGetValue("type", out var jType) ? (string)jType : null;
+                var errorMessage = error.TryGetValue("message", out var jMessage) ? (string)jMessage : null;
+                throw new OctoshiftCliException($"GraphQL response does not indicate success. Error Type: {errorType ?? "UNKNOWN"}, Error Message: {errorMessage ?? "UNKNOWN"}");
```
Comment (dylan-smith) [inline src/Octoshift/GithubApi.cs:630]: IMO we should make the error simpler. Probably just `new OctoshiftCliException(errorMessage ?? "UNKNOWN")`  If users want all the gory details they can look in the verbose logs.

**Review by dylan-smith — CHANGES_REQUESTED** [review]

File: src/Octoshift/GithubApi.cs (hunk 1 above)
Comment (ArinGhazarian) [reply src/Octoshift/GithubApi.cs]: addressed in c3064ce7d63ca11b181d2ab77bde4376b04f2472

**Review by dylan-smith — APPROVED** [review]

**Pull request merged**